Add release notes for 5.17 (#2109)
sensu/sensu-docs: content/sensu-go/5.17/release-notes.md

---

## 5.17.0 release notes

**January 23, 2020** &mdash; The latest release of Sensu Go, version 5.17.0, is now available for download.
PLACEHOLDER FOR RELEASE SUMMARY.

See the [upgrade guide][1] to upgrade Sensu to version 5.17.0.

**NEW FEATURES:**

> - ([Commercial feature][106]) Added [HTTP API for secrets provider management][108], and a built-in `Env` secrets provider. The secrets provider resource is implemented for checks, mutators, and handlers.

Author: ```suggestion
- ([Commercial feature][106]) Added [HTTP API for secrets provider management][108] and a built-in `Env` secrets provider. The secrets provider resource is implemented for checks, mutators, and handlers.
```

---

## 5.17.0 release notes

> **January 23, 2020** &mdash; The latest release of Sensu Go, version 5.17.0, is now available for download.

Author: ```suggestion
**January 27, 2020** &mdash; The latest release of Sensu Go, version 5.17.0, is now available for download.
```

> **January 27, 2020** &mdash; The latest release of Sensu Go, version 5.17.0, is now available for download.

Author: ```suggestion
**January 28, 2020** &mdash; The latest release of Sensu Go, version 5.17.0, is now available for download.
```

PLACEHOLDER FOR RELEASE SUMMARY.

See the [upgrade guide][1] to upgrade Sensu to version 5.17.0.

**NEW FEATURES:**

> - ([Commercial feature][106]) Added [HTTP API for secrets provider management][108] and a built-in `Env` secrets provider. The secrets provider resource is implemented for checks, mutators, and handlers.

Reviewer: We should mention that we also support Vault for secrets management.

> - ([Commercial feature][106]) Added [HTTP API for secrets provider management][108] and a built-in `Env` secrets provider. The secrets provider resource is implemented for checks, mutators, and handlers.

Author: ```suggestion
- ([Commercial feature][106]) Added [HTTP API for secrets management][108], with a built-in `Env` secrets provider and support for HashiCorp Vault secrets management. The secrets provider resource is implemented for checks, mutators, and handlers.
```

- Added the `keepalive-handlers` agent configuration flag to specify the keepalive handlers to use for an entity's events.

**IMPROVEMENTS:**

- ([Commercial feature][106]) Upgraded the size of the events auto-incremented ID in the PostgreSQL store to a 64-bit variant, which allows you to store many more events and avoids exhausting the sequence.
- In the [web UI][107], labels that contain URLs are now clickable links.

**FIXES:**

- ([Commercial feature][106]) Fixed a bug where the event check state was not present when using the PostgreSQL event store.
- Fixed a memory leak in the entity cache.
- Initialization via [`sensu-backend-init`][109] is now implemented for Docker.
- Fixed a bug that prevented `sensuctl entity delete` from returning an error when attempting to delete a non-existent entity.
- In the [web UI][107], fixed a bug that duplicated event history in the event timeline chart.

> - Sensuctl command assets installed via Bonsai now use the `sensuctl` namespace.

Author: ```suggestion
- `sensuctl command` assets installed via Bonsai now use the `sensuctl` namespace.
```

## 5.17.0 release notes

> **January 28, 2020** &mdash; The latest release of Sensu Go, version 5.17.0, is now available for download. This is a significant release, with new features, improvements, and fixes! We’re ecstatic to announce the release of secrets management, which eliminates the need to expose sensitive information in your Sensu configuration. When a Sensu component (e.g. check, handler, etc.) requires a secret (like a username or password), Sensu will be able to fetch that information from one or more external secrets providers (e.g. Hashicorp Vault) and provide it to the Sensu component via temporary environment variables. By leveraging secrets management, you’re able to move secrets out of your Sensu configuration, giving you the ability to safely, easily, and confidently share your Sensu configurations with your fellow Sensu users! This release also includes per entity keepalive event handler configuration, a sought after feature for users who have migrated from Sensu 1.x to Sensu Go.

Author: ```suggestion
**January 28, 2020** &mdash; The latest release of Sensu Go, version 5.17.0, is now available for download.
This is a significant release, with new features, improvements, and fixes!
We’re ecstatic to announce the release of secrets management, which eliminates the need to expose sensitive information in your Sensu configuration.
When a Sensu component (e.g. check, handler, etc.) requires a secret (like a username or password), Sensu will be able to fetch that information from one or more external secrets providers (e.g. HashiCorp Vault) and provide it to the Sensu component via temporary environment variables.
Secrets management allows you to move secrets out of your Sensu configuration, giving you the ability to safely and confidently share your Sensu configurations with your fellow Sensu users!
This release also includes per-entity keepalive event handler configuration, a sought-after feature for users who have migrated from Sensu 1.x to Sensu Go.
```
